lxd/storage/drivers/driver_powerflex_utils.go

```diff
@@ -1114,28 +1041,35 @@ func (d *powerflex) unmapVolume(vol Volume) error {
 
 	// Wait until the volume has disappeared.
 	volumePath, _, _ := d.getMappedDevPath(vol, false)
-	if volumePath != "" {
-		ctx, cancel := context.WithTimeout(d.state.ShutdownCtx, 10*time.Second)
-		defer cancel()
-
-		if !waitGone(ctx, volumePath) {
-			return fmt.Errorf("Timeout whilst waiting for PowerFlex volume to disappear: %q", vol.name)
-		}
+	if volumePath != "" && !connectors.WaitDiskDeviceGone(d.state.ShutdownCtx, volumePath, 10) {
+		return fmt.Errorf("Timeout whilst waiting for PowerFlex volume to disappear: %q", vol.name)
 	}
 
 	// In case of SDC the driver doesn't manage the underlying connection to PowerFlex.
 	// Therefore if this was the last volume being unmapped from this system
 	// LXD will not try to cleanup the connection.
-	if d.config["powerflex.mode"] == powerFlexModeNVMe {
+	if d.config["powerflex.mode"] == connectors.TypeNVME {
 		mappings, err := client.getHostVolumeMappings(host.ID)
 		if err != nil {
 			return err
 		}
 
 		if len(mappings) == 0 {
+			pool, err := d.resolvePool()
+			if err != nil {
+				return err
+			}
+
+			domain, err := d.client().getProtectionDomain(pool.ProtectionDomainID)
+			if err != nil {
+				return err
+			}
+
+			targetQN := domain.SystemID
+
 			// Disconnect from the NVMe subsystem.
 			// Do this first before removing the host from PowerFlex.
-			err := d.disconnectNVMeSubsys()
+			err = d.connector().Disconnect(targetQN)
```

Comment:
`DisconnectAll`?

Reply:
I would avoid DisconnectAll, as previously discussed, this would disconnect all NVMe volumes from the LXD host, which is not desired.